commit/merkleroot/outcome.go

```diff
@@ -43,57 +49,70 @@ func (p *Processor) getOutcome(
-		return Outcome{}, nextState
+		return Outcome{}, nextState, nil
 	}
 
 	switch nextState {
 	case selectingRangesForReport:
-		return reportRangesOutcome(q, p.lggr, consensusObservation, p.offchainCfg.MaxMerkleTreeSize, p.destChain), nextState
+		return reportRangesOutcome(q, p.lggr, consObservation, p.offchainCfg.MaxMerkleTreeSize, p.destChain),
+			nextState,
+			nil
 	case buildingReport:
 		if q.RetryRMNSignatures {
 			// We want to retry getting the RMN signatures on the exact same outcome we had before.
 			// The current observations should all be empty.
-			return previousOutcome, buildingReport
+			return previousOutcome, buildingReport, nil
 		}
-		return buildReport(q, p.lggr, consensusObservation, previousOutcome), nextState
+		return buildReport(q, p.offchainCfg.RMNEnabled, p.lggr, consObservation, previousOutcome), nextState, nil
 	case waitingForReportTransmission:
 		return checkForReportTransmission(
-			p.lggr, p.offchainCfg.MaxReportTransmissionCheckAttempts, previousOutcome, consensusObservation), nextState
+			p.lggr, p.offchainCfg.MaxReportTransmissionCheckAttempts, previousOutcome, consObservation), nextState, nil
 	default:
-		p.lggr.Warnw("Unexpected next state in Outcome", "state", nextState)
-		return Outcome{}, nextState
+		return Outcome{}, nextState, fmt.Errorf("unexpected next state in Outcome: %v", nextState)
 	}
 }
 
 // reportRangesOutcome determines the sequence number ranges for each chain to build a report from in the next round
 func reportRangesOutcome(
 	_ Query,
 	lggr logger.Logger,
-	consensusObservation consensusObservation,
+	consObservation consensusObservation,
 	maxMerkleTreeSize uint64,
 	dstChain cciptypes.ChainSelector,
 ) Outcome {
 	rangesToReport := make([]plugintypes.ChainRange, 0)
 
-	observedOnRampMaxSeqNumsMap := consensusObservation.OnRampMaxSeqNums
-	observedOffRampNextSeqNumsMap := consensusObservation.OffRampNextSeqNums
-	observedRMNRemoteConfig := consensusObservation.RMNRemoteConfig
+	observedOnRampMaxSeqNumsMap := consObservation.OnRampMaxSeqNums
+	observedOffRampNextSeqNumsMap := consObservation.OffRampNextSeqNums
+	observedRMNRemoteConfig := consObservation.RMNRemoteConfig
 
 	offRampNextSeqNums := make([]plugintypes.SeqNumChain, 0)
 
 	for chainSel, offRampNextSeqNum := range observedOffRampNextSeqNumsMap {
+		offRampNextSeqNums = append(offRampNextSeqNums, plugintypes.SeqNumChain{
+			ChainSel: chainSel,
+			SeqNum:   offRampNextSeqNum,
+		})
+
 		onRampMaxSeqNum, exists := observedOnRampMaxSeqNumsMap[chainSel]
 		if !exists {
 			continue
 		}
 
-		if offRampNextSeqNum <= onRampMaxSeqNum {
+		if onRampMaxSeqNum < offRampNextSeqNum-1 {
+			lggr.Errorw("sequence numbers between offRamp and onRamp reached an impossible state, "+
+				"offRamp latest executed sequence number is greater than onRamp latest executed sequence number",
+				"chain", chainSel, "onRampMaxSeqNum", onRampMaxSeqNum, "offRampNextSeqNum", offRampNextSeqNum)
+		}
```

Comment:
I guess we'd see this any time a node's RPC is out of date, maybe even each time the node is restarted. Do you think it makes sense to include a hint in the comment?